websockify/websocket.py

```diff
@@ -1,1109 +1,825 @@
 #!/usr/bin/env python
 
 '''
-Python WebSocket library with support for "wss://" encryption.
+Python WebSocket library
 Copyright 2011 Joel Martin
+Copyright 2016 Pierre Ossman
 Licensed under LGPL version 3 (see docs/LICENSE.LGPL-3)
 
 Supports following protocol versions:
     - http://tools.ietf.org/html/draft-ietf-hybi-thewebsocketprotocol-07
     - http://tools.ietf.org/html/draft-ietf-hybi-thewebsocketprotocol-10
     - http://tools.ietf.org/html/rfc6455
-
-You can make a cert/key with openssl using:
-openssl req -new -x509 -days 365 -nodes -out self.pem -keyout self.pem
-as taken from http://docs.python.org/dev/library/ssl.html#certificates
-
 '''
 
-import os, sys, time, errno, signal, socket, select, logging
-import array, struct
+import sys
+import array
+import errno
+import socket
+import struct
 from base64 import b64encode, b64decode
 
-# Imports that vary by python version
+try:
+    import numpy
+except ImportError:
+    import warnings
+    warnings.warn("no 'numpy' module, HyBi protocol will be slower")
+    numpy = None
 
-# python 3.0 differences
-if sys.hexversion > 0x3000000:
-    b2s = lambda buf: buf.decode('latin_1')
-    s2b = lambda s: s.encode('latin_1')
-    s2a = lambda s: s
-else:
-    b2s = lambda buf: buf  # No-op
-    s2b = lambda s: s      # No-op
-    s2a = lambda s: [ord(c) for c in s]
-try:    from io import StringIO
-except: from cStringIO import StringIO
-try:    from http.server import SimpleHTTPRequestHandler
-except: from SimpleHTTPServer import SimpleHTTPRequestHandler
+try:
+    import ssl
+except ImportError:
+    ssl = None
```

Comment:
We're still on 2.4 at Cendio, but that's partly because we haven't had a good excuse to raise the requirements. I can put in a separate commit that removes any fixes for pre-2.6.